### PR TITLE
Subprojects in merge commands

### PR DESCRIPTION
**Pull request by alex-mckenna**
This PR adds the ability to specify specific subprojects to deploy. This makes it possible to write things like

merge and deploy foo
merge and deploy foo, bar
merge and deploy foo, bar to environment
merge and deploy foo, bar on friday
Merge commits will include an additional line Deploy-Subprojects: with a value of either all, or a comma-separated list of the subprojects to deploy.

CC: @Riscky, the PR for this has changed to this new one

### CODE REVIEW

File: src/Project.hs [hunk 1]
```diff
@@ -169,12 +171,17 @@ data PullRequestStatus
 newtype DeployEnvironment = DeployEnvironment Text
   deriving (Eq, Show, Generic)
 
+data DeploySubprojects
+  = AllSubprojects
```
Comment (joris-burgers) [inline src/Project.hs:175]: `AllSubproject` sounds a bit confusing to me if projects don't have subprojects, but only 1. What do you think about `EntireProject`? There is no confusion then for edge-cases.

File: tests/ParserSpec.hs [hunk 2]
```diff
@@ -77,7 +78,23 @@ parserSpec = do
...
+        dummyParse "@bot merge and deploy aaa, bbb to production" `shouldBe`
+          Success (Approve $ MergeAndDeploy (OnlySubprojects ["aaa", "bbb"]) (DeployEnvironment "production"), AnyDay)
+
+      it "fails when an unknown subproject is specified" $
+        dummyParse "@bot merge and deploy ccc to production" `shouldBe`
+          -- I'm not super happy with this error message, it _should_ say that we expect to see
+          -- either 'to', or the name of a known subproject.
+          ParseError
+            "comment:1:23:\n  |\n1 | @bot merge and deploy ccc to production\n\
+            \  |                       ^^\nunexpected \"cc\"\nexpecting \"to\" or white space\n"
```
Comment (joris-burgers) [inline tests/ParserSpec.hs:97]: Can you add a test for `@bot merge and deploy aaa` without the environment, to see what it does.

**Review by joris-burgers — APPROVED** [review]
Some minor comments, but LGTM

File: src/Project.hs (hunk 1 above)
Comment (alex-mckenna) [reply src/Project.hs:175]: _Who am I to question a dutchman on the topic of bikesheds?_

File: tests/ParserSpec.hs (hunk 2 above)
Comment (alex-mckenna) [reply tests/ParserSpec.hs:97]: Two tests added:

  - That we can use subprojects with an implicit environment (`@bot merge and deploy aaa`)
  - That we can do the above with a merge window (`@bot merge and deploy aaa on friday`)

**Comment by Riscky** [comment]
Taking over the deployment from @alex-mckenna
@OpsBotPrime take the wheel

**Comment by HoffBotPrime** [comment]
Pull request approved for merge by @Riscky, rebasing now.

**Comment by HoffBotPrime** [comment]
[CI job :yellow_circle:](https://channable.semaphoreci.com/workflows/3efa2558-1bd4-4671-818c-7c94d2d1b98f?pipeline_id=53d722c9-4093-484b-ba72-49fbe20ab38f) started.
